src/components/package/show/package-show.js

```diff
@@ -81,14 +87,31 @@ class PackageShow extends Component {
     this.setState({ isCoverageEditable });
   };
 
+  handleSectionToggle = ({ id }) => {
+    this.setState((curState) => {
+      const newState = _.cloneDeep(curState);
+      newState.sections[id] = !newState.sections[id];
+      return newState;
+    });
+  }
+
+  handleExpandAll = (sections) => {
+    this.setState((curState) => {
+      const newState = _.cloneDeep(curState);
```

Comment:
Here you can use `lodash/fp/set`.

```js
import set from 'lodash/fp/set';

let next = set('sections', sections, this.state);
this.setState(next);
```

You could also just use spread operator here,

```js
this.setState({
  ...this.state,
  sections
});
```

Reply:
@taras: is there an advantage of doing one way versus the other? ```let next = set('sections', sections, this.state);
this.setState(next);``` vs.
```this.setState({
  ...this.state,
  sections
});```?

Reply:
I would say that you probably want to use the simplest way possible. In this case, the spread operator relies on something that's built into JavaScript so it's the simplest way.

Reply:
@taras: also, just to ensure my understanding between the old way and using the fp way: Using the new way is advantageous because we are saving on memory by not creating an entirely new object of state and replacing in existing state with the new value using fp..is that right?

Reply:
In React apps, it's generally good idea to practice doing things immutably. In this case, the implementation is correct but style can be better. The patterns around working with immutable data add up once you start recognizing them, so it's a good habit to start writing immutable code rather than using immutable patterns.